client/apis/objectstorage/v1alpha2/bucket_types.go

```diff
@@ -117,13 +140,18 @@ type BucketStatus struct {
 	ReadyToUse *bool `json:"readyToUse,omitempty"`
 
 	// bucketID is the unique identifier for the backend bucket known to the driver.
+	// Must be at most 2048 characters and consist only of alphanumeric characters ([a-z0-9A-Z]),
+	// dashes (-), dots (.), and underscores (_).
 	// +optional
 	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9._-]+$`
 	// +kubebuilder:validation:XValidation:message="boundBucketName is immutable once set",rule="self == oldSelf"
 	BucketID string `json:"bucketID,omitempty"`
```

Comment:
For bucket ID's, I'm trying to find the middle groud between a strong formal definition that might conflict with some edge cases between S3/Azure/GCS bucket naming -- versus totally unvalidated input.

Azure has the longest bucket ID limit at 1024 chars, and doubling that to 2048 seemed somewhat reasonable in case drivers want to add extra text based on other things.

Among the 3, the common character set is alphanums, dot, dash, and underscore. I opted to not get too complex here. Limiting chars to these should hinder any code injection type attacks a lot without risking unexpected compatibility in corner cases. 

WDYT @JoelSpeed ? @shanduur ?

Reply:
Are any of these case sensitive? Should we limit to lowercase alphanums? Otherwise that seems reasonable. Have you looked beyond the existing platforms to other future platforms at all?